src/pages/pass-finalisation/cat-be/components/code-78-present/code-78-present.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { TestCategory } from '../../../../../shared/models/test-category';
+import { Code78Present, Code78NotPresent } from '../../../../../modules/tests/pass-completion/pass-completion.actions';
+
+@Component({
+  selector: 'code-78-present',
+  templateUrl: 'code-78-present.html',
+})
+
+export class Code78PresentComponent implements OnChanges {
+  @Input()
+  form: FormGroup;
+
+  @Input()
+  category: String;
+
+  @Output()
+  code78Present = new EventEmitter<Code78Present>();
+
+  @Output()
+  code78NotPresent = new EventEmitter<Code78NotPresent>();
+
+  formControl: FormControl;
+  static readonly fieldName: string = 'code78PresentCtrl';
+
+  ngOnChanges(): void {
+    if (!this.formControl) {
+      this.formControl = new FormControl('', [Validators.required]);
+      this.form.addControl(Code78PresentComponent.fieldName, this.formControl);
+    }
+  }
+
+  shouldHideBanner(): boolean {
+    if (this.getTransmission()) {
+      return (
+        !this.formControl.valid || !this.getTransmission().dirty
+      );
+    }
+    return true;
+  }
+
+  isInvalid(): boolean {
+    return !this.formControl.valid && this.formControl.dirty;
+  }
+
+  getTransmission() {
+    return this.form.controls['transmissionCtrl'];
```

Comment:
this file shouldn't know anything about transmission, it should be passed in as an input (maybe the decision to show/hide the banner should be in the page above)